src/crypto.js

```diff
@@ -0,0 +1,283 @@
+   * @returns {CryptoKey} - The generated AES key.
+   */
+const genAESKey = (extractable, mode, keySize) => {
+  return window.crypto.subtle.generateKey({
+    name: mode || 'AES-GCM',
+    length: keySize || 128
+  }, extractable || true, ['decrypt', 'encrypt'])
+}
+
+/**
+  * Export a CryptoKey into a raw|jwk key
+  *
+  * @param {CryptoKey} key - The CryptoKey
+  * @param {string} [type] - The type of the exported key
+  * @returns {arrayBuffer|Object} - The raw key or the key as a jwk format
+  */
+const exportKey = (key, type = 'raw') => {
+  return window.crypto.subtle.exportKey(type, key)
+    .then(key => {
+      if (type === 'raw') return new Uint8Array(key)
+      return key
+    })
+}
+
+/**
+  * Import a raw|jwk as a CryptoKey
+  *
+  * @param {arrayBuffer|Object} key - The key
+  * @param {string} [type] - The type of the key to import ('raw', 'jwk')
+  * @param {string} [mode] - The mode of the key to import (default 'AES-GCM')
+  * @returns {arrayBuffer|Object} - The cryptoKey
+  */
+const importKey = (key, type = 'raw', mode = 'AES-GCM') => {
+  return window.crypto.subtle.importKey(type, key, { name: mode }
+    , true, ['encrypt', 'decrypt'])
+}
+
+/**
+ * Decrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to decrypt
+ * @param {ArrayBuffer} key - The AES key as raw data. 128 or 256 bits
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The decrypted buffer
+ */
+const decryptBuffer = (data, key, cipherContext) => {
+  // TODO: test input params
+  return window.crypto.subtle.decrypt(cipherContext, key, data)
+    .then(result => new Uint8Array(result))
+}
+
+/**
+ * Encrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to encrypt
+ * @param {ArrayBuffer} key - The AES CryptoKey
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The encrypted buffer
+ */
+const encryptBuffer = (data, key, cipherContext) => {
+  return window.crypto.subtle.encrypt(cipherContext, key, data)
+    .then(result => new Uint8Array(result))
+}
+
+/**
+ * Encrypt data
+ *
+ * @param {CryptoKey} key - The AES CryptoKey
+ * @param {string | Object} - The data to encrypt
+ * @param {string} [format] - The ciphertext and iv encoding format
+ * @returns {Object} - The stringified ciphertext object (ciphertext and iv)
+ */
+const encrypt = (key, data, format = 'hex') => {
```

Comment:
Keep the same order between key and data parameters in encrypt and encryptBuffer function

Reply:
ok done.